plugins/eos/gs-plugin-eos.c

```diff
@@ -2005,6 +2046,13 @@ gs_plugin_app_upgrade_download (GsPlugin *plugin,
 		break;
 	}
 
+	/* if the OS upgrade has been disabled */
+	if (priv->updater_proxy == NULL) {
+		g_set_error (error, GS_PLUGIN_ERROR, GS_PLUGIN_ERROR_FAILED,
+			     "the OS upgrade has been disabled in the EOS plugin");
+		return FALSE;
+	}
```

Comment:
Ditto, `get_os_upgrade()` will probably crash above when this is `NULL`

Reply:
See my reply above.